### PR TITLE
Add download tests.

### LINKED ISSUE #12: tests for `download_file`
## Summary

We need tests for `download_file`.

## Acceptance Criteria

Tests for `download_file` run on ci and pass

## Related to

#1

### PR DESCRIPTION
**Pull request by ericdatakelly**
Resolves #12

Changes proposed in this pull request:
- Adds two functions to test good and bad URLs

Related #1

### CODE REVIEW

**Review by rycrow — COMMENTED** [review]
these download tests look good. how will `download` be called? will we iterate across a list, calling download on each item? or would we change `download` to take in a list and have it download all the files inside of the function?

**Comment by ericdatakelly** [comment]
Thanks, @rycrow.  Do you mean `download_file`? 

Pytest will run all tests in the `tests` directory.  It looks for files starting with `test_` and then it looks for functions starting with `test_`.  It can also look for classes, but that hasn't been implemented here yet.

When Pytest runs, it will find this file and run the code that is inside it.
```python
def test_download_file_good_url():
    """Test downloading good URLs."""

    url = "https://conda.anaconda.org/conda-forge/noarch/backports-1.0-py_2.tar.bz2"
    f_name = download_file(url=url)

    assert f_name == "backports-1.0-py_2.tar.bz2", "Filename should match name in URL."

    f_path = Path(f_name).resolve()
    assert f_path.exists(), f"File should be written to {f_path}."

    size = f_path.stat().st_size
    assert 3000 < size < 4000, "File size should be around 3.6kB."

    # Remove the file (clean up)
    f_path.unlink()
```

Notice the function starts with `test_` and then uses the name of the function it tests `download_file` and then has additional info if  needed, in this case that it will test with a "good url".  Also notice that the file is independent of the other code that is being tested.  It included the function to be tested `download_file` and then only Python built in functions.  Other libraries are ok, but simple tests with few or no dependencies tend to be best.

File: ib_manifest_util/util.py [hunk 1]
```diff
@@ -75,18 +75,23 @@ def run_subprocess(command: str):
         sys.stdout.write(line.decode("utf-8"))
 
 
-def download_file(url: str):
+def download_file(url: str) -> str:
```
Comment (kcpevey) [inline ib_manifest_util/util.py]: I agree with @rycrow here, we should allow for a single url or a list of str to download multiple files.

**Review by kcpevey — APPROVED** [review]
One suggestion on accepting multiple urls for downloading. If you agree, that can be done in this PR or a separate one.

**Comment by ericdatakelly** [comment]
Ah, I misunderstood.  Thanks @kcpevey.

File: ib_manifest_util/util.py (hunk 1 above)
Comment (ericdatakelly) [reply ib_manifest_util/util.py]: I can add it to this PR.

Comment (ericdatakelly) [reply ib_manifest_util/util.py]: I added the changes. @rycrow @kcpevey

**Pull request merged**